sqlmesh/core/plan/definition.py

```diff
@@ -279,14 +303,26 @@ def loaded_snapshot_intervals(self) -> t.List[LoadedSnapshotIntervals]:
             if not snapshot.change_category:
                 continue
             loaded_snapshots.append(LoadedSnapshotIntervals.from_snapshot(snapshot))
-            for downstream_indirect in self.indirectly_modified.get(snapshot.name, set()):
-                downstream_snapshot = self.context_diff.snapshots[downstream_indirect]
+            for downstream_snapshot in self.indirectly_modified.get(snapshot, set()):
                 # We don't want to display indirect non-breaking since to users these are effectively no-op changes
                 if downstream_snapshot.is_indirect_non_breaking:
                     continue
                 loaded_snapshots.append(LoadedSnapshotIntervals.from_snapshot(downstream_snapshot))
         return loaded_snapshots
 
+    def _refresh_dag_and_ignored_snapshots(self) -> None:
+        self.ignored_snapshot_names: t.Set[str] = set()
+        self._dag: DAG[str] = DAG()
+        snapshots = self.context_diff.snapshots.values()
+        for name, snapshot in self.context_diff.snapshots.items():
```

Comment:
Seems like this iteration must happen in DAG order to ensure the correct behavior. Does `self.context_diff.snapshots` guarantee that order? Even if it does, I'm concerned that it's somewhat dangerous to assume that the dict is in correct order. I'd prefer this to be explicit.

Reply:
Yeah great catch I completely agree. I will enforce the order.

Reply:
https://github.com/TobikoData/sqlmesh/pull/1018/commits/b4bdfffb254d51b42e82e504c92b61b9372acb51